src/cdk/stepper/stepper.ts

```diff
@@ -84,6 +96,10 @@ export class CdkStepper {
   @Input()
   get selectedIndex() { return this._selectedIndex; }
   set selectedIndex(index: number) {
+    this._steps.toArray()[this._selectedIndex].interacted = true;
+    for (let i = 0; i < index; i++) {
```

Comment:
Add a comment here explaining why you return early if a preceding step is invalid